Geofencing/app/src/main/java/com/google/android/gms/location/sample/geofencing/MainActivity.java

```diff
@@ -410,20 +429,17 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
                 // again" prompts). Therefore, a user interface affordance is typically implemented
                 // when permissions are denied. Otherwise, your app could appear unresponsive to
                 // touches or interactions which have required permissions.
-                showSnackbar(R.string.permission_denied_explanation, R.string.settings,
-                        new View.OnClickListener() {
-                            @Override
-                            public void onClick(View view) {
-                                // Build intent that displays the App settings screen.
-                                Intent intent = new Intent();
-                                intent.setAction(
-                                        Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
-                                Uri uri = Uri.fromParts("package",
-                                        BuildConfig.APPLICATION_ID, null);
-                                intent.setData(uri);
-                                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                                startActivity(intent);
-                            }
+                showSnackbar(
+                        R.string.permission_denied_explanation,
+                        R.string.settings,
+                        view -> {
+                            // Build intent that displays the App settings screen.
```

Comment:
AFAIK showing the app details in system settings isn't a pattern we recommend for developers.